### PR TITLE
Benji/ios 272 add reply count to message

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: Shared/Extensions/UI Extensions/UIView+ScalingAnimation.swift [hunk 1]
```diff
@@ -34,4 +34,13 @@ extension UIView {
...
     }
+
+    @MainActor
+    func awaitAnimation(with duration: Theme.AnimationDuration, animations: @escaping () -> Void) async {
+        return await withCheckedContinuation { continuation in
+            UIView.animate(withDuration: duration.rawValue, animations: animations) { _ in
+                continuation.resume(with: .success(()))
+            }
+        }
+    }
```
Comment (MartinJibber) [inline Shared/Extensions/UI Extensions/UIView+ScalingAnimation.swift:45]: This is cool

File: iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift [hunk 2]
```diff
@@ -114,7 +114,14 @@ class SwipeableInputAccessoryView: View, UIGestureRecognizerDelegate {
                 self.updateInputType(with: first)
             }.store(in: &self.cancellables)
 
-        KeyboardManager.shared.$willKeyboardShow.mainSink { willShow in
+        KeyboardManager.shared.$willKeyboardShow
+            .filter({ willShow in
+                if let view = KeyboardManager.shared.inputAccessoryView as? SwipeableInputAccessoryView {
```
Comment (Benjamindodgson) [inline iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift:119]: Helps ensure we are looking at the correct input

File: iOS/Flows/Conversation/Cells/MessageStatusView.swift [hunk 3]
```diff
@@ -68,9 +71,23 @@ class MessageStatusView: UICollectionReusableView {
...
         self.addSubview(self.dateLabel)
         self.addSubview(self.statusLabel)
+
+        self.dateLabel.publisher(for: \.text).mainSink { _ in
+            self.layoutNow()
+        }.store(in: &self.cancellables)
+
+        self.statusLabel.publisher(for: \.text).mainSink { _ in
+            self.layoutNow()
+        }.store(in: &self.cancellables)
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/MessageStatusView.swift]: I think this is a memory leak.

**Review by MartinJibber — CHANGES_REQUESTED** [review]
I see a possible memory leak. Fix that and it's ready to merge I think.

**Pull request merged**